### PR TITLE
impl user error exception

### PR DESCRIPTION
**Pull request by andxu**
2. refactor the error handling for ResolveClasspathsHandler

### CODE REVIEW

**Comment by andxu** [comment]
https://github.com/Microsoft/java-debug/issues/55

**Comment by andxu** [comment]
https://github.com/Microsoft/java-debug/issues/85

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/UserErrorException.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+
+package com.microsoft.java.debug.core;
+
+public class UserErrorException extends DebugException {
+    private static final long serialVersionUID = -6001456457602995764L;
```
Comment (yaohaizh) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/UserErrorException.java]: What is this magic number for? Please comment it

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/UserErrorException.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    /**
+     * Create an user error exception indicates an user setting/operation is illegal by design.
```
Comment (yaohaizh) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/UserErrorException.java]: an ==> a

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/UserErrorException.java (hunk 1 above)
Comment (andxu) [reply com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/UserErrorException.java]: serialVersionUID is generated by eclipse using random number, it is used serialization orelse there will be a compilation warning.

Comment (yaohaizh) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/UserErrorException.java]: Naming convention for final static

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java [hunk 3]
```diff
@@ -32,53 +33,53 @@
...
-     * @throws CoreException
-     *             CoreException
+     * @throws DebugException when the project is not found or invalid.
      */
-    private static IJavaProject getJavaProjectFromName(String projectName) throws CoreException {
-        IJavaProject javaProject = JdtUtils.getJavaProject(projectName);
-        if (javaProject == null) {
-            throw new CoreException(new Status(IStatus.ERROR, JavaDebuggerServerPlugin.PLUGIN_ID,
-                    String.format("The project '%s' is not a valid java project.", projectName)));
+    private static IJavaProject getJavaProjectFromName(String projectName) throws DebugException {
```
Comment (yaohaizh) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java]: Could we make this as a Utility class?

Comment (andxu) [reply com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java]: It is in jinbo's change.

Comment (andxu) [reply com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java]: JdtUtils.getJavaProjectFromName  doesn't have the right error message for not-existing projects and non-java projects.

**Pull request closed without merging**